doc/pantry.md

```diff
@@ -0,0 +1,482 @@
+  version: '1.1'
+  sha256: c32231ff8548bccd4f3bafcc9b1eb84947a2e5e0897c50c048e0e7609fc443ce
+  pantry-tree:
+    size: 206
+    sha256: 79dbeddaf0fd507611687cefe9511c8fda489849fb0cac3894925716936290b2
+- size: 285152
+  subdir: wai
+  url: http://github.com/yesodweb/wai/archive/2f8a8e1b771829f4a8a77c0111352ce45a14c30f.zip
+  cabal-file:
+    size: 1717
+    sha256: 7b46e7a8b121d668351fa8a684810afadf58c39276125098485203ef274fd056
+  name: wai
+  version: 3.0.2.3
+  sha256: 3b6eb04f3763ca16432f3ab2135d239161fbe2c8811b8cd1778ffa67469289ba
+  pantry-tree:
+    size: 10296
+    sha256: ce431f1a22fcda89375ba5e35e53aee968eea23d1124fcba7cb9eae426daa2db
+- size: 285152
+  subdir: warp
+  url: http://github.com/yesodweb/wai/archive/2f8a8e1b771829f4a8a77c0111352ce45a14c30f.zip
+  cabal-file:
+    size: 6648
+    sha256: e3f01fd7417af923fd30962e9e6a4fe4de41ebc5e02af9819067fed79c9c6575
+  name: warp
+  version: 3.0.13.1
+  sha256: 3b6eb04f3763ca16432f3ab2135d239161fbe2c8811b8cd1778ffa67469289ba
+  pantry-tree:
+    size: 4292
+    sha256: d6b1def306a042b5fc500930302533a3ea828e916c99cbd82c0b7e2c4e3a8e09
+- size: 1442
+  filepath: acme-missiles-0.3.tar.gz
+  cabal-file:
+    size: 613
+    sha256: 2ba66a092a32593880a87fb00f3213762d7bca65a687d45965778deb8694c5d1
+  name: acme-missiles
+  version: '0.3'
+  sha256: e563d8b524017a06b32768c4db8eff1f822f3fb22a90320b7e414402647b735b
+  pantry-tree:
+    size: 226
+    sha256: 614bc0cca76937507ea0a5ccc17a504c997ce458d7f2f9e43b15a10c8eaeb033
+```
+
+## Snapshots
+
+_NOTE_ Stack has supported custom snapshots properly since version
+1.6. In version 1.11, the support for snapshots was moved to Pantry,
+and Stackage snapshots have moved over to using the same
+format. Therefore, there is no longer such a thing as "custom
+snapshots," there are simply "snapshots." Pantry snapshots follow the
+same format as Stack 1.6 "custom snapshots."
+
+Snapshots provide a list of packages to use, along with flags,
+ghc-options, and a few other settings. Snapshots may extend any other
+snapshot that can be specified in a `resolver` field. The packages
+specified follow the same syntax mentioned above for
+dependencies. Unlike `extra-deps`, however, no support for local
+directories is available in snapshots to ensure reproducibility.
+
+```yaml
+resolver: lts-8.21 # Inherits GHC version and package set
+compiler: ghc-8.0.1 # Overwrites GHC version in the resolver, optional
+
+name: my-snapshot # User-friendly name
+
+# Additional packages, follows extra-deps syntax
+packages:
+- unordered-containers-0.2.7.1
+- hashable-1.2.4.0
+- text-1.2.2.1
+
+# Override flags, can also override flags in the parent snapshot
+flags:
+  unordered-containers:
+    debug: true
+
+# Packages from the parent snapshot to ignore
+drop-packages:
+- wai-extra
+
+# Packages which should be hidden (affects script command's import
+# parser
+hidden:
+  wai: true
+  warp: false
+
+# Set GHC options for specific packages
+ghc-options:
+  warp:
+  - -O2
+```
```

Comment:
Haddock settings would go here, too? e.g. not to build a dependency with broken haddocks

Reply:
I haven't actually changed any of the behavior of `ghc-options` in this PR. I don't know if any of these settings would apply to Haddock or not. IIRC, there's an open issue about that, right?

Reply:
By here, I meant in this file, not this stanza — please excuse the ambiguity.

I can track down the issue — I'm not sure off the top of my head. But from the gitter chat:

> @dbaynard
> Is there a way to avoid haddock, or pass haddock arguments, when building a dependency, in the same way it is possible to set flags on dependencies? I'm unable to build some haddocks, but I want to ignore the error.
> …
> @snoyberg 
> Funny you should ask, we have the same requirement for Stackage. So I'm adding a new (hidden) curator: configuration that allows you to disable Haddocks, tests, and benchmarks on a package-by-package basis